pyspark/bigdl/keras/optimization.py

```diff
@@ -38,26 +40,80 @@ def to_bigdl_metrics(metrics):
 
     @staticmethod
     def to_bigdl_criterion(kloss):
-        # TODO: it may pass in an object and with parameters
-        if kloss == "categorical_crossentropy":
-            return bcriterion.ClassNLLCriterion()
-        elif kloss == "mse" or kloss == "mean_squared_error":
+        if isinstance(kloss, six.string_types):
+            kloss = kloss.lower()
+
+        if kloss == "categorical_crossentropy" or kloss == categorical_crossentropy:
+            return bcriterion.CategoricalCrossEntropy()
+        elif kloss == "mse" or kloss == "mean_squared_error" or kloss == mse:
             return bcriterion.MSECriterion()
-        elif kloss == "binary_crossentropy":
+        elif kloss == "binary_crossentropy" or kloss == binary_crossentropy:
             return bcriterion.BCECriterion()
-        elif kloss == "mae" or kloss == "mean_absolute_error":
+        elif kloss == "mae" or kloss == "mean_absolute_error" or kloss == mae:
             return bcriterion.AbsCriterion()
+        elif kloss == "hinge" or kloss == hinge:
```

Comment:
any difference?

Reply:
Yes. The former is a string and the latter is a function.

Reply:
ok